eds_scikit/plot/event_sequences.py

```diff
@@ -0,0 +1,226 @@
+        DataFrame gathering the events information. Must contain at least `person_id`, event, t_start and t_end columns.
+    event_col: Optional[str] = "event"
+        Column name of the events.
+    event_start_datetime_col: Optional[str] = "event_start_datetime"
+        Column name of the event start datetime.
+    event_end_datetime_col: Optional[str] = "event_end_datetime"
+        Column name of the event end datetime.
+    dim_mapping: Optional[Dict[str,Dict[str,Union[tuple(int),str]]]] = None
+        Mapping dictionary to provide plotting details on events. Must be of type :
+        ```python
+            dim_labelling = {
+                "event_1" : {"color":(255,200,150), "label":"Event 1"},
+                "event_2" : {"color":(200,255,150), "label":"Event 2"}
+            }
+        ```
+    index_date_col: Optional[str] = None
+        Column name of the index date to compute relative datetimes for events. For example, it could be the date of inclusion for each patient.
+    family_col: Optional[str] = None
+        Column name of family events. Events of a given family will be plot on the same row.
+    family_to_index: Optional[Dict[str,int]] = None
+        Dictionary mapping event family names to ordering indices.
+    list_person_ids: Optional[List[str]] = None
+        List of person_ids to plot. If None given, only the first three individual sequences will be plot.
+    same_x_axis_scale: Optional[bool] = False
+        Whether to use the same axis scale for all sequences.
+    subplot_height: Optional[int] = 200
+        Height of each plot.
+    subplot_width: Optional[int] = 500
+        Width of each plot.
+    point_size: Optional[int] = 400
+        Size of points for one-time events.
+    bar_height: Optional[int] = 20
+        Height of bars for continuous events.
+    title: Optional[str] = None
+        Chart title.
+    seed: int = 0
+        Seed to randomly draw colors when not provided.
+
+    Returns
+    -------
+    chart: alt.Chart
+        Chart with the plotted individual event sequences.
+    """
+    rng = np.random.RandomState(seed)
+
+    # Check required columns
+    required_columns = [
+        "person_id",
+        event_col,
+        event_start_datetime_col,
+        event_end_datetime_col,
+    ]
+
+    if index_date_col is not None:
+        required_columns.append(index_date_col)
+
+    if family_col is not None:
+        required_columns.append(family_col)
+
+    check_columns(df_events, required_columns=required_columns)
+
+    # Pre-selection of the sequences to plot and required columns.
+    if list_person_ids is None:
+        list_person_ids = list(df_events.person_id.unique()[:3])
+
+    order = {val: idx for idx, val in enumerate(list_person_ids)}
+    df_events = df_events.query("person_id in @list_person_ids").sort_values(
+        by="person_id", key=lambda x: x.map(order)
+    )
+
+    data_plot = df_events.sort_values(by=["person_id", event_start_datetime_col])
```

Comment:
Could you explain why we need a copy of `df_events` here? Also, we sort `data_plot` by `person_id` without using the order, and I wonder if this is the intended behavior since it can break the ordering.

In case you don't need a copy and if what you want is sorting by `person_id` with a custom order and then sorting with `event_start` I can suggest this:
```python
_, classes = np.unique(df_events["person_id"], return_inverse=True)
df_events["person_id_ordered"] = classes
df_events = (
    df_events.query("person_id in @list_person_ids")
    .sort_values(["person_id_ordered", event_start_datetime_col])
)
```

Reply:
I used to keep a copy of `df_events` in the development process as I returned the `data_plot` for quality checks, but you are raising a good point,it is not required anymore ! I will apply your suggestion, thanks.

Reply:
I think we should leave the same order as `list_person_id` though, to let the user the opportunity to provide the order they wish.

Reply:
Actually I think the sorting is not even required... I can just remove it.

Reply:
That looks great!